thorlcr/thorutil/thmem.cpp

```diff
@@ -2315,19 +2316,22 @@ class CThorAllocator : public CSimpleInterface, implements IThorAllocator, imple
             rowManager->setReleaseWhenModifyCallback(true, true);
         }
     }
-    CThorAllocator(IThorAllocator &sharedAllocator, unsigned channel)
+    CThorAllocator(CThorAllocator &sharedAllocator, unsigned channel)
     {
-        allocatorMetaCache.setown(createRowAllocatorCache(this));
+        allocatorMetaCache.set(sharedAllocator.queryAllocateCache());
         rowManager.set(sharedAllocator.queryRowManager()->querySlaveRowManager(channel));
         defaultFlags = sharedAllocator.queryFlags();
         logctx = sharedAllocator.queryLoggingContext();
         numChannels = 0;
+        dbgassertex(channel <= 0xffff);
```

Comment:
That may be too low in the medium term.  Is it the actually channel id?  if so it is already too low.